### PR TITLE
Feature/facet flow

### PR DESCRIPTION
**Pull request by DaQuirm**
Warp now uses proper authentication and so does Sandbox, yay! :smiley_cat: 
So it all revolves around `koa:sess` cookie and new Warp feature, `dynamic_entities`.
Full game flow should be working again, try `game.html` and `game_master.html`.

**Be sure to update game sessions in your DB**: add `game_master_id` (see GameSession schema).
Looks like we won't be needing that `role` property of the `User` anymore :cat: 

Oh and for @kirilknysh: there's now `user_data` cell which contains info from the OAuth provider (only `display_name` for now) :smile_cat:

### CODE REVIEW

File: app/facets/participant/game-master.coffee [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{nx} = require 'nexus-node'
+
+class GameMaster
+
+	constructor: (@participant) ->
+
+		@participant.game_master = @
+
+		@puzzles = new nx.Cell
+			'<<-*': [@participant.game_session, 'puzzles']
```
Comment (R1ZZU) [inline app/facets/participant/game-master.coffee]: we definitely need some docs on nexus-arrows 😿

Comment (DaQuirm) [inline app/facets/participant/game-master.coffee]: coming soon to nexus wiki: everything you didn't want to know about bindings :cat:

File: app/session-manager.coffee [hunk 2]
```diff
@@ -1,27 +1,34 @@
...
-			facet:     facet
-			transport: transport
-			entities:  facet.entities
+	create: (transport, cookies, done) ->
+		user = {}
+
+		session_cookie = cookies.find ({name}) ->
+			name is 'koa:sess'
+
+		passport_data = JSON.parse (new Buffer session_cookie.value, 'base64').toString 'utf8'
```
Comment (R1ZZU) [inline app/session-manager.coffee]: koa:sess cookie is not just passport data, but whole session, so variable name is bit confusing

File: app/game-session.coffee [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {nx} = require 'nexus-node'
 
-RoundPhase    = require 'cssqd-shared/models/round-phase'
+RoundPhase    = require '../shared/models/round-phase'
```
Comment (R1ZZU) [inline app/game-session.coffee]: why do you like this require method more? @kirilknysh have done great job [here](https://github.com/DaQuirm/css-quickdraw-redux/commit/f30837a78bdd18a846e2c31e5ef250294e39e7b2) :smirk_cat:

File: app/session-manager.coffee (hunk 2 above)
Comment (DaQuirm) [inline app/session-manager.coffee]: right you are sir

File: app/game-session.coffee (hunk 3 above)
Comment (DaQuirm) [inline app/game-session.coffee]: it didn't work for me for some reason... does it really work like webpack aliases?
summoning @kirilknysh

Comment (R1ZZU) [inline app/game-session.coffee]: seems like it [should work](https://docs.npmjs.com/files/package.json#local-paths) and it works fine for me

File: app/facets/participant/participant.coffee [hunk 4]
```diff
@@ -1,16 +1,23 @@
 {nx} = require 'nexus-node'
 
+GameRole = require '../../../shared/models/game-role'
```
Comment (R1ZZU) [inline app/facets/participant/participant.coffee]: cssqd-shared

Comment (DaQuirm) [inline app/facets/participant/participant.coffee]: right

File: app/common/models/game-session.coffee [hunk 5]
```diff
@@ -3,6 +3,7 @@ mongoose = require 'mongoose'
 Schema = mongoose.Schema
 
 GameSessionSchema = Schema
+	game_master_id: Schema.Types.ObjectId
```
Comment (kirilknysh) [inline app/common/models/game-session.coffee]: Shouldn't it be an array of `Id`-s?

File: public/dom-sandbox/app.coffee [hunk 6]
```diff
@@ -3,4 +3,5 @@ require 'nexus'
 AppViewModel = require './view-models/app-viewmodel'
 
 window.addEventListener 'load', ->
+	document.cookie = 'koa:sess=eyJwYXNzcG9ydCI6eyJ1c2VyIjoic2FuZGJveCJ9fQ=='
```
Comment (kirilknysh) [inline public/dom-sandbox/app.coffee]: Shouldn't we move it outside the repo? As it's done with secret login tokens?

File: package.json [hunk 7]
```diff
@@ -66,7 +66,7 @@
     "mongoose": "^4.2.7",
     "nconf": "^0.7.1",
     "nexus-node": "^0.4.21",
-    "nexus-warp": "0.0.6",
+    "nexus-warp": "git+https://github.com/DaQuirm/nexus-warp.git",
```
Comment (kirilknysh) [inline package.json]: Why link to repo? Changes come too often?

File: app/common/models/game-session.coffee (hunk 5 above)
Comment (DaQuirm) [inline app/common/models/game-session.coffee]: One game master to rule 'em all?
I don't know, let's keep it like that for now?

File: public/dom-sandbox/app.coffee (hunk 6 above)
Comment (DaQuirm) [inline public/dom-sandbox/app.coffee]: There's an issue about this: #45
We definitely should come up with something more secure :cat:

File: package.json (hunk 7 above)
Comment (DaQuirm) [inline package.json]: Yes, releases became too frequent :smile_cat: 
I think it's more practical to keep it like that for a while

**Pull request merged**